lib/synapse/haproxy.rb

```diff
@@ -719,10 +729,9 @@ def update_backends(watchers)
           unless cur_backends[watcher.name].include? backend_name
```

Comment:
Might be a bit clearer to use an `if` instead of an `unless` (and swap the two bodies around).